libraries/AP_XRCE_Client/AP_XRCE_Client.cpp

```diff
@@ -0,0 +1,118 @@
+    }
+    else {
+        this->reliable_in=uxr_create_input_reliable_stream(&this->session,this->input_reliable_stream,BUFFER_SIZE_UDP,STREAM_HISTORY);
+        this->reliable_out=uxr_create_output_reliable_stream(&this->session,this->output_reliable_stream,BUFFER_SIZE_UDP,STREAM_HISTORY);
+    }
+
+    return true;
+}
+bool AP_XRCE_Client::create()
+{
+
+    if(_csem->take(5)){
+        this->participant_id=uxr_object_id(0x01,UXR_PARTICIPANT_ID);
+        const char* participant_xml = "<dds>"
+                                        "<participant>"
+                                            "<rtps>"
+                                                "<name>AP_XRCE_Client</name>"
+                                            "</rtps>"
+                                        "</participant>"
+                                      "</dds>";
+        this->participant_req=uxr_buffer_create_participant_xml(&this->session,this->reliable_out,this->participant_id,0,participant_xml,UXR_REPLACE);
+
+        this->topic_id=uxr_object_id(0x01,UXR_TOPIC_ID);
+        const char* topic_xml = "<dds>"
+                                    "<topic>"
+                                            "<name>AP_INSTopic</name>"
+                                            "<dataType>AP_INS</dataType>"
+                                    "</topic>"
+                                "</dds>";
+        this->topic_req=uxr_buffer_create_topic_xml(&this->session,this->reliable_out,this->topic_id,this->participant_id,topic_xml,UXR_REPLACE);
+    
+        this->pub_id=uxr_object_id(0x01,UXR_PUBLISHER_ID);
+        const char* pub_xml = "";
+        this->pub_req = uxr_buffer_create_publisher_xml(&this->session,this->reliable_out,this->pub_id,this->participant_id,pub_xml,UXR_REPLACE);
+
+        this->dwriter_id = uxr_object_id(0x01,UXR_DATAWRITER_ID);
+        const char* dwriter_xml = "<dds>"
+                                    "<data_writer>"
+                                        "<topic>"
+                                            "<kind>NO_KEY</kind>"
+                                            "<name>AP_INSTopic</name>"
+                                            "<dataType>AP_INS</dataType>"
+                                        "</topic>"
+                                    "</data_writer>"
+                                "</dds>";
+        this->dwriter_req = uxr_buffer_create_datawriter_xml(&this->session,this->reliable_out,this->dwriter_id,this->pub_id,dwriter_xml,UXR_REPLACE);
+
+        uint16_t requests[4] = {this->participant_req,this->topic_req,this->pub_req,this->dwriter_req};
+    
+        if(!uxr_run_session_until_all_status(&this->session,1000,requests,this->status,4)){
+            return false;
+        }
+        _csem->give();
+        return true;
+    }
+    return false;
+}
+
+void AP_XRCE_Client::write()
+{
+    if(_csem->take(5)){
+        if(this->connected)
+        {
```

Comment:
Avoid indenting

```suggestion
    if(!_csem->take(5)){
        return;
    }
   if(!this->connected)
       return;
   }
```

Reply:
Everything is formatted with `astyle` on the PR.